src/backend/metal/src/device.rs

```diff
@@ -114,10 +115,14 @@ fn get_final_function(
     })?;
 
     if !function_specialization {
-        assert!(
-            specialization.data.is_empty() && specialization.constants.is_empty(),
-            "platform does not support specialization",
-        );
+        if caps_support_debug_markers {
```

Comment:
the specialization is not related to the debug markers

Reply:
So how do i check if specialization is supported? That assertion fails on my machine, but i don't think it should even be being tested.

Reply:
The capability of checking for specialization support is already passed in as another parameter - `function_specialization`. Perhaps, all we want to do here is to change the `assert` into `error!` as you did, just without involving `caps_support_debug_markers`.

Reply:
Ah, that would work; the problem was that that assert was panicking on my machine, which doesn't support specialization apparently, but the examples work if i comment it out.